### PR TITLE
FIX : only plot eeg, mag, grad in evoked plots

### LINKED ISSUE #2407: example broken due to viz changes
See eg plot_read_evoked.py

get : KeyError: 'eog'

@jaeilepp I suppose it's due to some recent changes you made. thanks for taking a look.

### PR DESCRIPTION
**Pull request by agramfort**
fixes #2407

### CODE REVIEW

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
     scalings = _handle_default('scalings', scalings)
     titles = _handle_default('titles', titles)
     units = _handle_default('units', units)
-    channel_types = set(key for d in [scalings, titles, units] for key in d)
+    channel_types = ['eeg', 'mag', 'grad']  # valid types
     channel_types = sorted(channel_types)  # to guarantee consistent order
```
Comment (jaeilepp) [inline mne/viz/evoked.py]: You can remove sorting, if you change it to ``['eeg', 'grad', 'mag']``.

Comment (agramfort) [inline mne/viz/evoked.py]: ok but are you ok with this change?

Comment (jaeilepp) [inline mne/viz/evoked.py]: Yeah looks good.

**Comment by agramfort** [comment]
let's merge when travis is happy.

There are still some examples failing due to include/exclude mess with 'bad'

plot_shift_evoked.py, plot_eog_artifact_histogram.py cc @choldgraf

**Pull request merged**